app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/PurchaseDetailsController.scala

```diff
@@ -38,19 +38,19 @@ class PurchaseDetailsController @Inject()(
                                            formProvider: PurchaseDetailsFormProvider,
                                            repo: DeclarationJourneyRepository,
                                            view: PurchaseDetailsView
-                                          )(implicit ec: ExecutionContext, appConfig: AppConfig)
+                                         )(implicit ec: ExecutionContext, appConfig: AppConfig)
   extends DeclarationJourneyUpdateController with CurrencyConversionConnector {
 
   val form: Form[PurchaseDetailsInput] = formProvider()
 
   val onPageLoad: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
     // TODO replace with parameterised :idx, use headOption for single goods journey
-    request.declarationJourney.goodsEntries.headOption match {
+    request.declarationJourney.goodsEntries.entries.headOption match {
       case Some(goodsEntry) =>
 
         getCurrencies().map { currencyPeriod =>
-          val preparedForm = goodsEntry.maybePriceOfGoods match {
-            case Some(priceOfGoods) => form.fill(priceOfGoods.toPurchaseDetailsInput)
+          val preparedForm = goodsEntry.maybePurchaseDetails match {
+            case Some(purchaseDetails) => form.fill(purchaseDetails.purchaseDetailsInput)
             case None => form
           }
```

Comment:
few nested pattern matching, maybe extracting into smaller methods would help readability?

Reply:
I think these could all be re-factored to use fold:

val preparedForm = goodsEntry.maybePurchaseDetails fold(form) { purchaseDetails =>
            form.fill(purchaseDetails.purchaseDetailsInput)
}

there's a bunch that I could do on a follow-up pr?